nisystemlink/clients/notebook/_notebook_client.py

```diff
@@ -0,0 +1,356 @@
+import io
+from typing import List, Optional
+
+from nisystemlink.clients import core
+from nisystemlink.clients.core._uplink._base_client import BaseClient
+from nisystemlink.clients.core._uplink._file_like_response import (
+    file_like_response_handler,
+)
+from nisystemlink.clients.core._uplink._methods import (
+    delete,
+    get,
+    post,
+    put,
+    response_handler,
+)
+from nisystemlink.clients.core.helpers._iterator_file_like import IteratorFileLike
+from requests.models import Response
+from uplink import Part, Path
+
+from . import models
+
+
+def _query_executions_response_handler(response: Response) -> List[models.Execution]:
+    if response is None:
+        return []
+
+    executions = response.json()
+
+    return executions
+
+
+def _simple_response_handler(response: Response) -> dict:
+    if response is None:
+        return dict()
+
+    return response.json()
+
+
+class NotebookClient(BaseClient):
+    def __init__(self, configuration: Optional[core.HttpConfiguration] = None):
+        """Initialize an instance.
+
+        Args:
+            configuration: Defines the web server to connect to and information about
+                how to connect. If not provided, the
+                :class:`HttpConfigurationManager <nisystemlink.clients.core.HttpConfigurationManager>`
+                is used to obtain the configuration.
+
+        Raises:
+            ApiException: if unable to communicate with the Notebook Service.
+        """
+        if configuration is None:
+            configuration = core.HttpConfigurationManager.get_configuration()
+        super().__init__(configuration, base_path="/")
+
+    @get("ninotebook/v1/notebook/{id}")
+    def get_notebook(self, id: str) -> models.NotebookMetadata:
+        """Gets a notebook metadata by ID.
+
+        Args:
+            id: The ID of the notebook to read.
+
+        Returns:
+            The notebook metadata.
+
+        Raises:
+            ApiException: if unable to communicate with the ``/ninotebook`` service or provided invalid
+                arguments.
+        """
+        ...
+
+    @put("ninotebook/v1/notebook/{id}")
+    def _update_notebook(
```

Comment:
Private functions should be led with double underscore
```suggestion
    def __update_notebook(
```